### PR TITLE
Adding in local storage component to JWT token

### PR DESCRIPTION
**Pull request by taygrave**
in preparation for using AWS auth services which don't query cookies

also adding support for refreshing a token

plus tests!

### CODE REVIEW

**Comment by taygrave** [comment]
cc @lanetix/engineers

**Comment by todesignandconquer** [comment]
:yellow_heart:

File: src/index.js [hunk 1]
```diff
@@ -42,6 +49,31 @@ module.exports = (options) => {
...
+    if (expDate && expDate < new Date() && options.refresh) {
+      options.refresh()
+      .then(tokenStore.setToken.bind(tokenStore))
+    }
+  }
+
+  refreshToken()
+  const refreshInterval = options.refreshInterval
+  ? options.refreshInterval
+  : (60 * 1000)
```
Comment (sahiltalwar88) [inline src/index.js]: tabbing. also, could just write 60000 :P

**Comment by sahiltalwar88** [comment]
:heartbeat:

File: src/index.js [hunk 2]
```diff
@@ -23,10 +24,16 @@ const decodeToken = token => {
...
   options = extend({ cookie: 'XSRF-TOKEN' }, options)
 
-  let token = cookie.get && cookie.get(options.cookie)
-  let user = decodeToken(token)
+  let token
+  if ('localStorageKey' in options) {
+    token = ls.get(options.localStorageKey)
+  } else {
+    token = cookie.get && cookie.get(options.cookie)
+  }
```
Comment (XieX) [inline src/index.js]: Everything's better as a ternary operator

```javascript
token = options.localStorageKey ? ls.get(options.localStorageKey) : cookie.get && cookie.get(options.cookie)
```

But no really, it's fine the way it is ^_^

**Pull request merged**